### ISSUE TITLE
Controller from angular-ui-mention library not working

### ISSUE THREAD

**Issue by dieguito151** [meanjs/mean#1530]
I'm using Angular 1.5.5 and I'm trying to implement a mentions feature. I've installed `angular-ui-mention` (https://github.com/angular-ui/ui-mention) with Bower, added the library path to `/config/env/all.js`. I'm using the following directive: 

```
angular.module('core').directive('myMention', [
  function() {
    return {
      require: 'uiMention',
      link: function(scope, element, attrs, uiMention) {
        uiMention.label = function(choice) {
          return choice.first_name + ' ' + choice.last_name;
        };

        uiMention.findChoices = function() {
          return [];
        };
      }
    };
  }
]);

```
`angular-ui-mention` has the `uiMention` controller defined in the library, but my directive can't seem to find it, and Angular is throwing this error:

`angular.js:13920 Error: [$compile:ctreq] Controller 'uiMention', required by directive 'pdcMention', can't be found!`

Everything seems to be fine, and I can't figure out how to make it work. Is there any extra step I should do?

**Comment by dieguito151** [comment]
I did notice that the app server needs to be restarted for new libraries to be included in the source code. But yes, I did restart my server and it still doesn't work. Do I have to register the module, or include the path somewhere else?

**Comment by hyperreality** [comment]
I tried adding the directive to a base MEAN project, I had no problems. Let's go through a few things:
1. Why did you add the library path to `/config/env/all.js`? Are you using a super old version of MEAN.js or something? For about a year now the library path goes in `/config/assets/default.js`
2. It looks like you've got the syntax for writing the directive very wrong. 

Take a look at the correct example on the ui-mention page:
    .directive('myMention', function($http){
      return {
        require: 'uiMention',
A function is taking the http parameter and returning the directive. Also check out the page title directive in the MEAN project for an example of using the correct syntax that you can adapt.

Instead you've put the function inside square brackets, which is where you should put angular dependencies, not functions.

**Comment by dieguito151** [comment]
1. Indeed, I'm using an old version of MEAN.js. I'm working on a project that started almost 3 years ago (I have no idea what version this MEAN.js is), and migrating to a new version would prove to be time-consuming.

2. I just tried it that way as well and it's still not working. The problem seems to be that angular doesn't recognize the `angular-ui-mention` script, and not recognizing the `uiMention` controller.

I didn't include $http mainly because we use services/factories instead of $http/$resource as best practice, and also because I just wanted to verify that the directive is working, and would just populate dummy date in `return []`; I was going to add the service later. Also, we define all our dependencies in square brackets for minification.

What did you do to set up MEAN with the library? Install angular-ui-mention through bower, add the path to `/config/assets/default.js`, and set up the directive like the example? Pretty much that's what I did, maybe I'm missing a step or setting up something else.
